gui_app.py

```diff
@@ -496,6 +495,33 @@ def run_installer(self, evt=None):
     def reinstall_deps(self, evt=None):
         utils.check_dependencies()
 
+    def set_file_dialog_environment(self):
+        desktop_environment = os.environ.get('XDG_CURRENT_DESKTOP', '').lower()
+    
+        if 'gnome' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['GTK_FILE_CHOOSER_ACTION'] = 'open'
+        elif 'kde' in desktop_environment:
+            os.environ['TK_SILENCE_DEPRECATION'] = '1'
+            os.environ['KDE_FULL_SESSION'] = 'true'
+
+    def open_file_dialog(self, filetype_name, filetype_extension):
+        root = Tk()
+        root.withdraw()
+    
+        file_path = fd.askopenfilename(
+            title=f"Select {filetype_name}",
+            filetypes=[(filetype_name, f"*.{filetype_extension}"), ("All Files", "*.*")],
+        )
+
+        return file_path
```

Comment:
I don't think the `root = Tk()` and `root.withdraw()` lines are useful here, if we're content with using `fd.askopenfilename()`.

A hack that I found to hide hidden folders and files:
```python
try:
    # Dummy option below means the function throws an exception, which keeps the window from
    # opening but allows the variables to be set afterwards.
    self.root.tk.call('tk_getOpenFile', '-dummyoption')
except TclError:
    pass
self.root.tk.call('set', '::tk::dialog::file::showHiddenBtn', '1')
self.root.tk.call('set', '::tk::dialog::file::showHiddenVar', '0')

# Then call the fd.askopenfilename() function...
```
It looks like that code could be worked into this `open_file_dialog` function, and then the function could become a module-level function outside of the `ControlWindow` class. That would allow it to be reused in other cases. It might also be useful to have an analogous `open_folder_dialog` function to go with it.